control-plane/connect-inject/webhook/consul_dataplane_sidecar.go

```diff
@@ -383,3 +417,12 @@ func (w *MeshWebhook) sidecarResources(pod corev1.Pod) (corev1.ResourceRequireme
 
 	return resources, nil
 }
+
+// useProxyHealthCheck returns true if the pod has the annotation 'consul.hashicorp.com/use-proxy-health-check'
+// set to "true".
+func useProxyHealthCheck(pod corev1.Pod) bool {
+	if v, ok := pod.Annotations[constants.AnnotationUseProxyHealthCheck]; ok || v == "true" {
```

Comment:
It'd be nice to parse the boolean here so that other `True`-y values will be accepted too like we do for other boolean annotations e.g. `consul.hashicorp.com/transparent-proxy`